### ISSUE TITLE
Potential inconsistency in root profile parameterization

### ISSUE THREAD

**Issue by rosiealice** [NGEET/fates#290]
Looking through the parameter list, I noted there are two parameter types for defining the root profile. 

One of them (roota_par and rootb_par) is used to define the root profile for water stress (patch%rootfr_ft) at the very bottom of https://github.com/NGEET/fates/blob/master/biogeochem/EDPatchDynamicsMod.F90

The other (rootprof_beta) is used to calculate the input of root litter into the different soil layers in https://github.com/NGEET/fates/blob/master/biogeochem/EDPhysiologyMod.F90 on line 1552.

My vote would be to only use one of these, and ideally the simpler Jackson term (rootprof_beta), but which one is less important than being consistent. Is there a reason that we are preserving both that I don't know about?  CLM moved from the 2-parameter to 1-parameter model in the last couple of years, which is probably how this arose.

**Comment by rgknox** [comment]
From what I am following in this thread, it seems there is no argument for keeping two different parameter sets correct?

I am in the process of cleaning up some parameters right now and would like some resolution here.

I propose the following as a the definitive way to define the fine root profile.

```
  real(r8), allocatable :: rootprof_mode              ! Fine root profile functional form
  real(r8), allocatable :: rootprof_a(:)              ! Fine root profile scaling parameter A
  real(r8), allocatable :: rootprof_b(:)              ! Fine root profile scaling parameter B
```

Note that the rooprof_mode will allow us to choose between a generic 1 parameter exponential, 2 parameter exponential, Jackson's method and Zeng's method.  This would also allow us add alternative functions that normalize these profiles to a maximum rooting depth that is based on plant size stature.

Currently, we do use the rootprof beta for things like biomass turnover and respiration, but we use the roota and rootb parameters for all things hydraulics.  I suppose we would need to choose a default (Zeng? Jackson?)

**Comment by dlawrenncar** [comment]
I don't really understand all the uses of rooting distribution across CLM
and FATES, but during CLM5 development, we made an effort to remove
inconsistencies in rooting distribution across different components of the
model (water was using Zeng, carbon using Jackson).  It seems like this
might be a situation where you might want to consider having the control on
rooting distribution to be maintained within the host land model and we
could agree on a default rooting distribution when running CLM(FATES) or
ELM(FATES).

Perhaps at the same time, we should consider what would happen if we moved
to a dynamic rooting distribution, which is partially functional in CLM and
maybe more fully functional in ELM.

**Comment by rosiealice** [comment]
Hi @dlawrenncar and @rgknox ... 

This is a very old thread!   Thanks for delving into it again Ryan. 

With the BGC coupling of FATES, the rooting profile in FATES controls the inputs of the C/N(P) fluxes of varying decomposabilities into the soil profile, so in principle, there shouldn't be any definition of root profile in the host model at all... In addition, we want there to be the capacity for different PFTs to have different depth roots. 

A seperate issue is a little bit whether we need to keep the two parameter Zeng model. As I recall in CLM5 we had ditched it in favour of entirely Jackson-based profiles. Either way, we should definitely make it so that the code uses the same one for both functions. I do not  think it is/was intentional at all that the biomass code uses the Jackson scheme in particular, and this could be changed to match the HYDRO usage without being problematic.  

I can see that there -might- be a reason to have different rooting profiles for different functions if we thought that, say, the nutrient foraging roots near the surface had limited hydraulic functionality but a) I don't think I've seen empirical evidence of that sort of thing and b) if we wanted to do that we should do it deliberately than what is there now?

Incidentally, I dug up lots of grass roots today and they were all in the top 8cm, which was surprising, giving how crazy hot it is here in summer.   I hate it when nature fails to conform to my low dimensional optimality theories...

**Comment by dlawrenncar** [comment]
Hi @rosiealice,

Not sure I follow your comment about the host land model not needing to know the root profile at all.  It will need to know for hydrology.  So, I guess I still wonder if it would make sense for the rooting profile to be set by the host land model so that can ensure consistency between host land model and FATES.

**Comment by rosiealice** [comment]
So, FATES calculates the fraction of extraction at each root level, which the HLM can then use extract water once it knows the transpiration... Is there another application of the root profile that hydrology needs (infiltration?)

**Comment by dlawrenncar** [comment]
Oh.  Right.  Then no, I don't think the host model hydrology needs to know
about roots.

**Comment by rgknox** [comment]
fixed via #659
